autoPyTorch/evaluation/fit_evaluator.py

```diff
@@ -0,0 +1,317 @@
+        self.logger.info("Starting fit {}".format(split_id))
+
+        pipeline = self._get_pipeline()
+
+        train_split, test_split = self.splits[split_id]
+        assert test_split is None
+        self.Y_actual_train = self.y_train[train_split]
+        y_train_pred, y_valid_pred, y_test_pred = self._fit_and_predict(pipeline, split_id,
+                                                                        train_indices=train_split,
+                                                                        test_indices=test_split,
+                                                                        add_pipeline_to_self=True)
+        train_loss = self._loss(self.y_train[train_split], y_train_pred)
+        if y_valid_pred is not None:
+            loss = self._loss(self.y_valid, y_valid_pred)
+        elif y_test_pred is not None:
+            loss = self._loss(self.y_test, y_test_pred)
+        else:
+            loss = train_loss
+
+        additional_run_info = pipeline.get_additional_run_info() if hasattr(
+            pipeline, 'get_additional_run_info') else {}
+
+        status = StatusType.SUCCESS
+
+        self.logger.debug("In train evaluator fit_predict_and_loss, num_run: {} loss:{}".format(
+            self.num_run,
+            loss
+        ))
+        self.finish_up(
+            loss=loss,
+            train_loss=train_loss,
+            valid_pred=y_valid_pred,
+            test_pred=y_test_pred,
+            additional_run_info=additional_run_info,
+            file_output=True,
+            status=status,
+            opt_pred=None
+        )
+
+    def _fit_and_predict(self, pipeline: BaseEstimator, fold: int, train_indices: Union[np.ndarray, List],
+                         test_indices: None,
+                         add_pipeline_to_self: bool
+                         ) -> Tuple[np.ndarray, Optional[np.ndarray], Optional[np.ndarray]]:
+
+        X = {'train_indices': train_indices,
+             'val_indices': test_indices,
+             'split_id': fold,
+             'num_run': self.num_run,
+             **self.fit_dictionary}  # fit dictionary
+        y = None
+        fit_and_suppress_warnings(self.logger, pipeline, X, y)
+        self.logger.info("Model fitted, now predicting")
+        (
+            Y_train_pred,
+            Y_valid_pred,
+            Y_test_pred
+        ) = self._predict(
+            pipeline,
+            train_indices=train_indices,
+        )
+
+        if add_pipeline_to_self:
+            self.pipeline = pipeline
+
+        return Y_train_pred, Y_valid_pred, Y_test_pred
+
+    def _predict(self, pipeline: BaseEstimator,
+                 train_indices: Union[np.ndarray, List]
+                 ) -> Tuple[np.ndarray, Optional[np.ndarray], Optional[np.ndarray]]:
+
+        train_pred = self.predict_function(subsampler(self.X_train, train_indices), pipeline,
+                                           self.y_train[train_indices])
+
+        if self.X_valid is not None:
```

Comment:
Would we ever have a value for `x_valid `since this is a `fit_evaluator` only for the `no_resampling_strategy`

Reply:
but this X_valid is if the user provides a validation set. So separate from the train set if a user wants to provide a valid set, they can do so here

Reply:
Okay, so this is if the user maybe wants to know how much it changed compared for example to a previous run when he ran on only `X_train`, compared to running on `X_train + X_valid`

Reply:
yeah or just having a validation set, I don't really think there's much point to it. but for now I think its okay. we may need to decide whether to keep it later for the library. As it is not gonna be used to report epoch wise statistics yet

Reply:
True, we can change it in the future.